pyro/infer/mcmc/util.py

```diff
@@ -405,13 +407,14 @@ def initialize_model(model, model_args=(), model_kwargs={}, transforms=None, max
 
     pe_maker = _PEMaker(model, model_args, model_kwargs, trace_prob_evaluator, transforms)
 
-    # Note that we deliberately do not exercise jit compilation here so as to
-    # enable potential_fn to be picklable (a torch._C.Function cannot be pickled).
-    init_params = _find_valid_initial_params(model, model_args, model_kwargs, transforms,
-                                             pe_maker.get_potential_fn(), prototype_params,
-                                             num_chains=num_chains, init_strategy=init_strategy)
+    if initial_params is None:
+        # Note that we deliberately do not exercise jit compilation here so as to
+        # enable potential_fn to be picklable (a torch._C.Function cannot be pickled).
+        initial_params = _find_valid_initial_params(model, model_args, model_kwargs, transforms,
```

Comment:
We can move `prototype_params` also inside this block.

Reply:
Good catch, thanks!